### PR TITLE
land detector: small refactor while studying

### PR DESCRIPTION
**Pull request by MaEtUgR**
just naming and large number constants, should have no change in functionality!

### CODE REVIEW

File: src/modules/land_detector/MulticopterLandDetector.h [hunk 1]
```diff
@@ -81,19 +81,19 @@ class MulticopterLandDetector : public LandDetector
 		param_t maxClimbRate;
 		param_t maxVelocity;
 		param_t maxRotation;
-		param_t maxThrottle;
+		param_t minThrottle;
 		param_t minManThrottle;
-		param_t acc_threshold_m_s2;
+		param_t ff_acc_threshold;
```
Comment (bkueng) [inline src/modules/land_detector/MulticopterLandDetector.h]: how about something like `freefall_acc_threshold`?

File: src/modules/land_detector/MulticopterLandDetector.cpp [hunk 2]
```diff
@@ -187,7 +188,7 @@ bool MulticopterLandDetector::_get_landed_state()
 		// falling consider it to be landed. This should even sustain
 		// quite acrobatic flight.
 		if ((_min_trust_start > 0) &&
-		    (hrt_elapsed_time(&_min_trust_start) > 8 * 1000 * 1000)) {
+		    (hrt_elapsed_time(&_min_trust_start) > 8e6)) {
```
Comment (bkueng) [inline src/modules/land_detector/MulticopterLandDetector.cpp]: I just realized that 8e6 is actually a double, so you are changing the type here (and the other places as well).

Comment (LorenzMeier) [inline src/modules/land_detector/MulticopterLandDetector.cpp]: Yep, that's an issue. Please revert.

File: src/modules/land_detector/LandDetector.h [hunk 3]
```diff
@@ -127,10 +127,10 @@ class LandDetector
 	static constexpr uint32_t LAND_DETECTOR_UPDATE_RATE_HZ = 50;
 
 	/* Time in us that landing conditions have to hold before triggering a land. */
-	static constexpr uint64_t LAND_DETECTOR_TRIGGER_TIME_US = 2000000;
+	static constexpr uint64_t LAND_DETECTOR_TRIGGER_TIME_US = 2e6;
```
Comment (LorenzMeier) [inline src/modules/land_detector/LandDetector.h]: This is changing to a double, which then gets casted to an uint64_t.

File: src/modules/land_detector/MulticopterLandDetector.cpp [hunk 4]
```diff
@@ -178,7 +179,7 @@ bool MulticopterLandDetector::_get_landed_state()
 
 	// Return status based on armed state and throttle if no position lock is available.
 	if (_vehicleLocalPosition.timestamp == 0 ||
-	    hrt_elapsed_time(&_vehicleLocalPosition.timestamp) > 500000 ||
+	    hrt_elapsed_time(&_vehicleLocalPosition.timestamp) > 0.5e6 ||
```
Comment (LorenzMeier) [inline src/modules/land_detector/MulticopterLandDetector.cpp]: This leads to a double executed in soft-float (longer runtime, more stack). Should be reverted.

File: src/modules/land_detector/MulticopterLandDetector.cpp (hunk 2 above)
Comment (LorenzMeier) [inline src/modules/land_detector/MulticopterLandDetector.cpp]: Yes. please revert that change.

File: src/modules/land_detector/MulticopterLandDetector.h (hunk 1 above)
Comment (MaEtUgR) [inline src/modules/land_detector/MulticopterLandDetector.h]: I totally agree, it was just adapted to the naming of the other parameter but I'll also change that.

File: src/modules/land_detector/MulticopterLandDetector.cpp (hunk 2 above)
Comment (MaEtUgR) [inline src/modules/land_detector/MulticopterLandDetector.cpp]: you are totally right, I only tried the literal alone
`(uint64_t) 8e6` would do the trick
not so nice again, sorry then

File: src/modules/land_detector/LandDetector.h (hunk 3 above)
Comment (MaEtUgR) [inline src/modules/land_detector/LandDetector.h]: if you cast it to an integer explicitly the compiler does never output a double in the assembly

**Comment by MaEtUgR** [comment]
I like your thorough reviews, I can learn a lot. Thanks.

**Comment by bkueng** [comment]
@MaEtUgR I don't like these long literals either. C++14 will help, it introduces digit separators: https://en.wikipedia.org/wiki/C++14#Digit_separators

**Comment by MaEtUgR** [comment]
@bkueng thanks for the info, that's a good idea. But I would like a literal ending like 1e6f or 1e6d to tell the compiler it's an integer.

**Pull request merged**